### ISSUE TITLE
custom_dataset error

### ISSUE THREAD

**Issue by KazuyaMishio** [meituan/YOLOv6#1020]
### Before Asking

- [X] I have read the [README](https://github.com/meituan/YOLOv6/blob/main/README.md) carefully. 我已经仔细阅读了README上的操作指引。

- [X] I want to train my custom dataset, and I have read the [tutorials for training your custom data](https://github.com/meituan/YOLOv6/blob/main/docs/Train_custom_data.md) carefully and organize my dataset correctly; (FYI: We recommand you to apply the config files of xx_finetune.py.) 我想训练自定义数据集，我已经仔细阅读了训练自定义数据的教程，以及按照正确的目录结构存放数据集。（FYI: 我们推荐使用xx_finetune.py等配置文件训练自定义数据集。）

- [X] I have pulled the latest code of main branch to run again and the problem still existed. 我已经拉取了主分支上最新的代码，重新运行之后，问题仍不能解决。

### Search before asking

- [X] I have searched the YOLOv6 [issues](https://github.com/meituan/YOLOv6/issues) and found no similar questions.

### Question

I want to train my dataset. However, I get the following error. Please help.
`python tools/train.py --batch-size 7 --conf-file configs/yolov6l_finetune.py --data-path data/dataset.yaml --epoch 20`
<img width="782" alt="スクリーンショット (22)" src="https://github.com/meituan/YOLOv6/assets/158554685/1e52f5e9-ba6e-4e76-98c4-14ee0a0ebb56">
environment
python 3.8.0
Package                      Version
---------------------------- --------------------
absl-py                      2.1.0
addict                       2.4.0
astunparse                   1.6.3
cachetools                   5.3.2
certifi                      2023.11.17
charset-normalizer           3.3.2
colorama                     0.4.6
contourpy                    1.1.1
cycler                       0.12.1
et-xmlfile                   1.1.0
flatbuffers                  23.5.26
fonttools                    4.47.2
gast                         0.4.0
gitdb                        4.0.11
GitPython                    3.1.41
google-auth                  2.26.2
google-auth-oauthlib         1.0.0
google-pasta                 0.2.0
grpcio                       1.60.0
h5py                         3.10.0
idna                         3.6
importlib-metadata           7.0.1
importlib-resources          6.1.1
keras                        2.13.1
kiwisolver                   1.4.5
libclang                     16.0.6
logging3                     0.0.2
Markdown                     3.5.2
markdown-it-py               3.0.0
MarkupSafe                   2.1.3
matplotlib                   3.7.4
mdurl                        0.1.2
natsort                      8.4.0
numpy                        1.24.3
oauthlib                     3.2.2
onnx                         1.15.0
onnx-simplifier              0.4.35
opencv-python                4.9.0.80
openpyxl                     3.1.2
opt-einsum                   3.3.0
packaging                    23.2
Pillow                       9.5.0
pip                          24.0
protobuf                     4.25.2
psutil                       5.9.8
pyasn1                       0.5.1
pyasn1-modules               0.3.0
pycocotools                  2.0.7
Pygments                     2.17.2
pynvml                       11.5.0
pyparsing                    3.1.1
python-dateutil              2.8.2
PyYAML                       6.0.1
requests                     2.31.0
requests-oauthlib            1.3.1
rich                         13.7.0
rsa                          4.9
scipy                        1.10.1
setuptools                   69.0.3
six                          1.16.0
smmap                        5.0.1
tensorboard                  2.13.0
tensorboard-data-server      0.7.2
tensorflow                   2.13.0
tensorflow-estimator         2.13.0
tensorflow-intel             2.13.0
tensorflow-io-gcs-filesystem 0.31.0
termcolor                    2.4.0
thop                         0.1.1.post2209072238
torch                        1.8.1+cu111
torchaudio                   0.8.1
torchvision                  0.9.1+cu111
tqdm                         4.66.1
typing_extensions            4.5.0
urllib3                      2.1.0
Werkzeug                     3.0.1
wheel                        0.42.0
wrapt                        1.16.0
zipp                         3.17.0
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 517.00       Driver Version: 517.00       CUDA Version: 11.7     |
|-------------------------------+----------------------+----------------------+
| GPU  Name            TCC/WDDM | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|  Memory-Usage | GPU-Util  Compute M. |
|                                                   |                            |               MIG M.       |
|===============================+==============|
|   0  NVIDIA GeForce ... WDDM  | 00000000:01:00.0 Off |                  N/A   |
| N/A   43C    P0    23W /  N/A |      0MiB /  6144MiB |      0%      Default    |
|                                                 |                                 |                  N/A        |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                                                                 |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory    |
|        ID   ID                                                   Usage                                    |
|==============================================|
|  No running processes found                                                                   |
+-----------------------------------------------------------------------------+

**Comment by Chilicyy** [comment]
Hi @KazuyaMishio , we added cache-ram option to speed up training, please make sure that you have pulled the latest code. More details of cache-ram can be found [here](https://github.com/meituan/YOLOv6/commit/d512ce7c4f103e8887960198505518bed404abdc#diff-d838e62affabf730936455bc1acface59ad6c2d409bd0e9e08ee0fb3fb898392).
